url49526af437_Maria_G_CS325_tgz-pJ8-assignments_farmer_FarmerImplJ8/src/frs/hotgammon/common/GameImpl.java

```diff
@@ -0,0 +1,387 @@
+	    		new Placement(Color.BLACK,Location.B8),
+	    		// B12 = 5 red
+	    		new Placement(Color.RED,Location.B12),
+	    		new Placement(Color.RED,Location.B12),
+	    		new Placement(Color.RED,Location.B12),
+	    		new Placement(Color.RED,Location.B12),
+	    		new Placement(Color.RED,Location.B12),
+	    		// R12 = 5 black
+	    		new Placement(Color.BLACK,Location.R12),
+	    		new Placement(Color.BLACK,Location.R12),
+	    		new Placement(Color.BLACK,Location.R12),
+	    		new Placement(Color.BLACK,Location.R12),
+	    		new Placement(Color.BLACK,Location.R12),		
+	    		// R8 = 3 red
+	    		new Placement(Color.RED,Location.R8),
+	    		new Placement(Color.RED,Location.R8),
+	    		new Placement(Color.RED,Location.R8),		
+	    		// R6 = 5 red
+	    		new Placement(Color.RED,Location.R6),
+	    		new Placement(Color.RED,Location.R6),
+	    		new Placement(Color.RED,Location.R6),
+	    		new Placement(Color.RED,Location.R6),
+	    		new Placement(Color.RED,Location.R6),		
+	    		// R1 = 2 black
+	    		new Placement(Color.BLACK,Location.R1),
+	    		new Placement(Color.BLACK,Location.R1),
+	    });
+	  
+	//Notify Observers
+	  for( GameObserver gO : this.observers ){
+		  gO.setStatus("New Game: Click the dice to roll. Player with the highest dice value goes first!");
+		  gO.diceRolled(new int[]{1,1});
+	  }
+
+  }
+  
+  public void nextTurn() {
+
+	  //Check For Winner
+	  if(isWinner()){
+		  return;
+	  }
+	  
+	  playerInTurn = turnDeterminer.nextTurn();
+	  
+	  //Roll Dice
+	  diceRollDeterminer.rollDice();
+	  int[] dRoll = diceRollDeterminer.getDiceRoll();
+	  
+	  if(turns == 0){
+		  while(dRoll.length > 2){
+			  diceRollDeterminer.rollDice();
+			  dRoll = diceRollDeterminer.getDiceRoll();
+		  }
+		  playerInTurn = determineStartingPlayer(dRoll);
+		
+	  }
+	  
+	  //Create diceRoll 
+	  if(dRoll.length == 2){
+		  diceRoll = new ArrayList<Integer>(Arrays.asList(dRoll[0], dRoll[1])) ;
+	  }
+	  else{
+		  diceRoll = new ArrayList<Integer>(Arrays.asList(dRoll[0], dRoll[1], dRoll[2], dRoll[3])) ;
+	  }
+	  
+	  turns++;
+	  movesLeft = diceRoll.size();
+	  
+	  String statusMessage = "";
+	  if(turns == 1){
+		  statusMessage = playerInTurn.toString() + " rolled the highest value! ";
+	  }
+	  
+	  //Notify Observers
+	  for( GameObserver gO : this.observers ){
+		  gO.diceRolled(diceThrown());
+		  gO.setStatus(statusMessage + getPlayerInTurn().toString() + " has "+ getNumberOfMovesLeft() +" moves left...");
+	  }
+	  
+  }
+  
+  private Color determineStartingPlayer(int[] dRoll){
+	  return (dRoll[0] > dRoll[1]) ? Color.RED : Color.BLACK;
+  }
+  
+  private void moveDuringConfigure(Location from, Location to){
+	  Color checkerColor = (from == Location.R_BEAR_OFF) ? Color.RED : Color.BLACK;
+	  gameBoard.move(from, to, checkerColor);
+	  
+	  //Notify Observers
+	  for( GameObserver gO : this.observers ){
+		  gO.checkerMove(from, to);
+	  }
+  }
+  public boolean move(Location from, Location to) { 
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 182-233 into a separate method.